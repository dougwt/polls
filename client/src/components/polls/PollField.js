import React from 'react';
import { Row, Input } from 'react-materialize';

const PollField = ({
  input,
  label,
  icon,
  s = 12,
  meta: { active, error, touched }
}) => {
  return (
    <Row>
      <Input
        label={touched && error ? error : label}
        labelClassName={touched && error ? 'red-text' : ''}
        icon={icon}
        {...input}
        validate
        s={s}
      />
    </Row>
  );
};

export default PollField;
