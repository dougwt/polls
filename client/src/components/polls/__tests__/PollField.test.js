import React from 'react';
import { render } from 'enzyme';
import PollField from '../PollField';

describe('PollField', () => {
  const props = {
    input: true,
    label: 'Sample label',
    icon: null,
    meta: {
      active: null,
      error: null,
      touched: null
    }
  };

  const pollField = render(<PollField {...props} />);

  it('renders properly', () => {
    expect(pollField).toMatchSnapshot();
  });

  it('contains a label', () => {
    expect(pollField.find('label').length).toBe(1);
    expect(pollField.find('label').text()).toEqual(props.label);
  });

  it('contains an input', () => {
    expect(pollField.find('input').length).toBe(1);
  });
});
