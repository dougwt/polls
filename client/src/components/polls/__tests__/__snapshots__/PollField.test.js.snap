// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollField renders properly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "row",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "col input-field s12",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "validate",
              "id": "input_0",
              "type": "text",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "",
                "for": "input_0",
              },
              "children": Array [
                Object {
                  "data": "Sample label",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "for": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "for": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
              "type": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "",
              "for": "input_0",
            },
            "children": Array [
              Object {
                "data": "Sample label",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "validate",
                "id": "input_0",
                "type": "text",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "type": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "for": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "for": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
